index.bs

```diff
@@ -329,36 +329,58 @@ When this method is invoked, it MUST run the following steps:
   1. Let |promise| be [=a new Promise=] in the [=relevant realm=] of this {{XRSystem}}.
   1. If |mode| is {{XRSessionMode/"inline"}}, [=/resolve=] |promise| with <code>true</code> and return it.
   1. If the requesting document's origin is not allowed to use the "xr-spatial-tracking" [[#permissions-policy|permissions policy]], [=reject=] |promise| with a "{{SecurityError}}" {{DOMException}} and return it.
-  1. Run the following steps [=in parallel=]:
-    1. Let |device| be the result of [=ensure an immersive XR device is selected|ensuring an immersive XR device is selected=].
-    1. If |device| is null, [=/resolve=] |promise| with <code>false</code> and abort these steps.
-    1. If |device|'s [=list of supported modes=] does not [=list/contain=] |mode|, [=queue a task=] to [=/resolve=] |promise| with <code>false</code> and abort these steps.
-    1. [=queue a task=] to [=/resolve=] |promise| with <code>true</code>.
+  1. Check whether the session |mode| is supported as follows:
+    <dl class="switch">
+      : If the user agent and system are known to [=never support=] |mode| sessions
+      :: [=/Resolve=] |promise| with <code>false</code>.
+
+      : If the user agent and system are is known to [=usually support=] |mode| sessions
+      :: |promise| MAY be [=/resolved=] with <code>true</code>.
+
+      : Otherwise
+      :: Run the following steps [=in parallel=]:
+        1. Let |device| be the result of [=ensure an immersive XR device is selected|ensuring an immersive XR device is selected=].
+        1. If |device| is null, [=/resolve=] |promise| with <code>false</code> and abort these steps.
+        1. If |device|'s [=list of supported modes=] does not [=list/contain=] |mode|, [=queue a task=] to [=/resolve=] |promise| with <code>false</code> and abort these steps.
+        1. The user agent SHOULD ensure [=user intent=] to allow pages to know XR capabilities are available is well undertood.
```

Comment:
How about

> The user agent SHOULD ensure user intent to allow fingerprinting in cases where `isSessionSupported()` represents a fingerprinting vector, as detailed below

Reply:
That's easier to parse, definitely. The nature of the switch statement implies that you'll only hit this case in situations where isSessionSupported() _does_ "represent a fingerprinting vector", so we could possibly trim that wording, but I don't think it's problematic to leave it in either.

I haven't been happy with the way this sentence (my version or yours) doesn't describe the step to be taken when fingerprinting is blocked, though. What would you think about negating the wording to read like this:

> If user intent to avoid advertising their system's XR capabilities is well understood queue a task to resolve `promise` with `false` and abort these steps.

WDYT?

Reply:
Seems good!

Reply:
Great! Updated.